sparrow-py/pysrc/sparrow_py/session.py

```diff
@@ -1,53 +1,83 @@
-from sparrow_py import _ffi
-from sparrow_py import Expr
 from typing import Optional
-import pyarrow as pa
+
 import pandas as pd
+import pyarrow as pa
+from sparrow_py import Expr
+from sparrow_py import _ffi
+
 
-class InvalidSchemaException(Exception):
+class UndefinedFieldError(Exception):
     def __init__(self, message):
         super().__init__(message)
 
 
 class Table(Expr):
     """ "A table expression."""
-    
+
     data: list[pa.Table] = []
 
-    def __init__(self, 
+    def validate_table(
```

Comment:
I think this should be `@classmethod` then (since it's not a valid method on the `Table`).